numpy/lib/_arraysetops_impl.py

```diff
@@ -364,6 +366,193 @@ def _unique1d(ar, return_index=False, return_inverse=False,
     return ret
 
 
+# Array API set functions
+
+class UniqueAllResult(NamedTuple):
+    values: np.ndarray
+    indices: np.ndarray
+    inverse_indices: np.ndarray
+    counts: np.ndarray
+
+
+class UniqueCountsResult(NamedTuple):
+    values: np.ndarray
+    counts: np.ndarray
+
+
+class UniqueInverseResult(NamedTuple):
+    values: np.ndarray
+    inverse_indices: np.ndarray
+
+
+UniqueAllResult.__module__ = "numpy"
+UniqueCountsResult.__module__ = "numpy"
+UniqueInverseResult.__module__ = "numpy"
```

Comment:
I don't believe these named tuples are actually available in the main namespace (as `__module__` now claims), no? This could potentially cause issue I imagine with anything relying on `__module__` + `__name__` for retrieving an object, _e.g._ does pickling even work now?

Reply:
Good point - @rgommers Do you think that `namedtuple` classes of return objects for setops should be present in the main namespace? Now when I think about it they shouldn't and I would leave `__module__` as `numpy.lib._arraysetops`.

We had similar discussion for index tricks, e.g. `np.r_` which is `<numpy.lib._index_tricks_impl.RClass at 0x102fd6b20>`. `RClass` return class is in a private module only.

Reply:
I don't think those should be public. But yes, pickling should work, and adding a test to confirm that would be useful. So `__module__` should just not be set, then it should be automatically correct.